### DISCUSSION TITLE
How to generate adoc files included in manual.adoc file?

### DISCUSSION THREAD

**Discussion by lcheylus** [rust-lang/rust-analyzer#16940] [Q&A] [answered]
I'm trying to package `rust-analyzer` with its documentation.

During build of a release, HTML documentation is built from `docs/user/manual.adoc` file (with `asciidoctor` tool). `manual.adoc` file includes 4 other files (`adoc` format): 
- `generated_config.adoc`
- `generated_features.adoc`
- `generated_assists.adoc`
- `generated_diagnostic.adoc`

How to generate these 4 `adoc` files during build with `cargo` ?

**Comment by flodiebold** [comment, marked as answer]
As far as I understand,
- `generated_config.adoc` is checked into the repo and there's a test that makes sure it's up to date (`cargo test -p rust-analyzer -- generate_config_documentation`);
- `diagnostics` and `assists` are generated by `cargo xtask codegen`;
- `features` are generated by `cargo test -p rust-analyzer -- sourcegen_feature_docs` (and _not_ checked in).

Not sure why they're so different, it might just be for historical reasons.

**Reply by Veykril** [reply]
They differ because I haven't gotten around to migrating the rest to `xtask` (I've done this incrementally)

**Reply by lcheylus** [reply]
Thanks, the 4 adoc files are correctly generated via `cargo test` / `cargo xtask`.
